### ISSUE TITLE
Is WebAssembly supported now?

### ISSUE THREAD

**Issue by ErikXu** [redis-rs/redis-rs#508]
I am trying to use `redis-rs` in wasm, but I got an `I/O error: operation not supported on this platform` error.

Here is my code refering the [basic.rs](https://github.com/mitsuhiko/redis-rs/blob/master/examples/basic.rs):

```
use redis::{Client, RedisResult, cmd, FromRedisValue, Cmd};

#[no_mangle]
pub unsafe extern fn do_redis_code(url: &str) -> redis::RedisResult<()> {
    let client = redis::Client::open(url)?;
    let mut con = client.get_connection()?;

    Ok(())
}

#[no_mangle]
pub unsafe extern "C" fn run() {
    match do_redis_code("redis://xxx.xxx.xxx.xxx:6379/") {
        Err(err) => {
            println!("Could not execute example:");
            println!("{}: {}", err.category(), err);
        }
        Ok(()) => {println!("Success");}
    }
}
```

Cargo.toml:

```
[package]
name = "redis-trial"
version = "0.1.0"
authors = []
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]

[dependencies.redis]
version = "*"

[lib]
path="src/lib.rs"
crate-type=["cdylib"]
```

Here is the error information:

```
...
Could not execute example:
I/O error: operation not supported on this platform
...
```

The build command:

```
cargo build --target wasm32-wasi --release
wasmtime target/wasm32-wasi/release/redis_trial.wasm --invoke run
```

**Comment by dlcrista** [comment]
Were you able to figure out a solution to this?

**Comment by ErikXu** [comment]
@dlcrista Not really. I tried to build a rust and implement the socket feature refer the following links:
https://radu-matei.com/blog/towards-sockets-networking-wasi/
https://engineering.linecorp.com/en/blog/adding-experimental-webassembly-support-to-decaton-part-1/

But I failed mainly due to the source code version mismatched.

As I was writing a istio/envoy wasm filter, I used rest api communicating with redis instead.

**Comment by dlcrista** [comment]
It looks like it works for me now

**Comment by dlcrista** [comment]
Well, I am able to include the following in my toml 
`[dependencies.redis]
version = "*"
`
and I am able to compile with `cargo build --target wasm32-unknown-unknown --release`

but I see an error in the logs: `"operation not supported on this platform"` when I try to do `con.get_connection()`. Maybe it's an error in my code?

**Comment by ErikXu** [comment]
It's not the problem of your code. This error message is output by rust as it does not implement socket in wasi32.

**Comment by 247gzs** [comment]
@ErikXu  How did you solve this problem

**Comment by ErikXu** [comment]
@247gzs Not solved in wasi32. Maybe you can try to call a rest api which communicates with redis instead.

**Comment by 247gzs** [comment]
@ErikXu This is a good suggestion, thank you.

**Comment by nihohit** [comment]
sync WASM build is now enabled after https://github.com/redis-rs/redis-rs/pull/1818
